nixos/modules/services/mail/roundcube.nix

```diff
@@ -228,13 +228,15 @@ in
       (mkIf (cfg.database.host == "localhost") {
         requires = [ "postgresql.service" ];
         after = [ "postgresql.service" ];
-        path = [ config.services.postgresql.package ];
       })
       {
         after = [ "network-online.target" ];
         wantedBy = [ "multi-user.target" ];
+
+        path = [ config.services.postgresql.package ];
         script = let
-          psql = "${lib.optionalString (!localDB) "PGPASSFILE=${cfg.database.passwordFile}"} ${pkgs.postgresql}/bin/psql ${lib.optionalString (!localDB) "-h ${cfg.database.host} -U ${cfg.database.username} "} ${cfg.database.dbname}";
+          postgresBinary = if cfg.database.host == "localhost" then "psql" else "${pkgs.postgresql}/bin/psql";
```

Comment:
my case is `services.postgresql.package` is customized to `postgresql_14_jit` which isn't the default, but before this patch `pkgs.postgresql` would duplicate `14_jit` and make my system closure bigger,

with this diff 14_jit's psql is used for roundcube's stuff which makes it happy since there's no extra postgres closure emanating from roundcube. maybe it's also better for protocol compat.

what difference are you proposing? this is noticeable enough for me as i found this while pushing a new system closure and wondering why there's so much postgres

Reply:
> my case is services.postgresql.package is customized to postgresql_14_jit which isn't the default, but before this patch pkgs.postgresql would duplicate 14_jit and make my system closure bigger,

I think there's a misunderstanding: I fully agree with what you're saying. However, the `if`/`else` for `postgresBinary` doesn't seem needed. Just set it to `psql` and you're good since `config.services.postgres.package` is already in path (and the option has a default, so it also exists if you're using a remote database and no local postgresql).

Reply:
ohhh! that makes sense. sorry for the misunderstanding, sleepy :PP 

pushed a fixed version now. there's like 3 other modules/packages that need the same treatment but that's some other time

Reply:
Ehh, no worries at all!
Looking good now, thanks!